test/protobuf/field_options_processor_test.exs

```diff
@@ -0,0 +1,79 @@
+defmodule Protobuf.FieldOptionsProcessorTest do
+  @moduledoc false
+
+  use ExUnit.Case, async: true
+
+  alias Protobuf.FieldOptionsProcessor
+
+  test "type_to_spec String.t and StringValue" do
+    extype = "String.t"
+    assert FieldOptionsProcessor.type_to_spec(:TYPE_MESSAGE, "Google.Protobuf.StringValue", false, [extype: extype]) ==
+      extype <> " | nil"
+  end
+
+  test "type_to_spec String.t() and StringValue" do
+    extype = "String.t()"
+    assert FieldOptionsProcessor.type_to_spec(:TYPE_MESSAGE, "Google.Protobuf.StringValue", false, [extype: extype]) ==
+      extype <> " | nil"
+  end
+
+  test "type_to_spec repeated" do
+    extype = "String.t()"
+    assert FieldOptionsProcessor.type_to_spec(:TYPE_MESSAGE, "Google.Protobuf.StringValue", true, [extype: extype]) ==
+      "[String.t() | nil]"
+  end
+
+  test "type_to_spec invalid extype" do
+    extype = "integer"
+    assert_raise RuntimeError, "Invalid extype pairing, " <>
+      "integer not compatible with Google.Protobuf.StringValue",
+      fn ->
+        FieldOptionsProcessor.type_to_spec(:TYPE_MESSAGE, "Google.Protobuf.StringValue", false, [extype: extype])
+      end
+  end
+
+  test "type_to_spec invalid type" do
+    extype = "String.t()"
+    assert_raise RuntimeError,
+      "Sorry Google.Protobuf.UnrealValue does not support the field option extype",
+      fn ->
+        FieldOptionsProcessor.type_to_spec(:TYPE_MESSAGE, "Google.Protobuf.UnrealValue", false, [extype: extype])
+      end
+  end
+
+  test "type_default" do
+
+    assert is_nil(FieldOptionsProcessor.type_default(Google.Protobuf.BoolValue, extype: "boolean"))
+
+    assert is_nil(FieldOptionsProcessor.type_default(Google.Protobuf.Timestamp, extype: "DateTime.t()"))
+    assert is_nil(FieldOptionsProcessor.type_default(Google.Protobuf.Timestamp, extype: "DateTime.t"))
+    assert is_nil(FieldOptionsProcessor.type_default(Google.Protobuf.Timestamp, extype: "NaiveDateTime.t()"))
+
+    # Typo
+    assert_raise RuntimeError, "Invalid extype pairing, Datetime.t not compatible with " <>
+      "Elixir.Google.Protobuf.Timestamp. Supported types are DateTime.t() or NaiveDateTime.t()",
+      fn -> FieldOptionsProcessor.type_default(Google.Protobuf.Timestamp, extype: "Datetime.t") end
+  end
+
+  test "encoding and decoding timestamp" do
+    dt = DateTime.utc_now()
```

Comment:
wouldn't hurt to test a different timezone

Reply:
good catch, I added tests. It is tricky, we always load utc even if the input timezone was something else